articles/112_generated_interfaces_cpp.md

```diff
@@ -28,16 +30,25 @@ This article specifies the generated C++ code for ROS interface types defined in
 ## Namespacing
 
 All code of a ROS package should be defined in a namespace named after the package.
-To separate the generated code from other code within the package it is defined in a sub namespace:
+To separate the generated code from other code within the package it is defined in a sub-namespace:
+
 * namespace for ROS messages: `package_name::msg`.
 * namespace for ROS services: `package_name::srv`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** Those namespaces are not consistent with ROS 1 usage.
+</div>
+
```

Comment:
> If we don't do this should we change Python to not put them in submodules, e.g. from std_msgs import String instead of from std_msgs.msg import String?

The Python representation is outside the scope of this C++ article. I have not seen any similar Python design document, yet. 

When we do have one, I will probably advocate doing it similarly to the way Python works in ROS 1.